### PR TITLE
feature/consumer/retailer-list

### PR DESCRIPTION
**Pull request by Kwang-Kian-Hui**
Display retailer list on consumer home page.  Only feature, UI not complete.

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 include: package:lint/analysis_options.yaml
 
 analyzer:
+  exclude:
+    - "**/*.g.dart"
+    - "**/*.freezed.dart"
```
Comment (sinhaugoh) [inline analysis_options.yaml]: what is the purpose of doing this?

File: lib/consumer/home/presentation/drawer_widget.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/material.dart';
```
Comment (sinhaugoh) [inline lib/consumer/home/presentation/drawer_widget.dart]: this file should be under core/presentation as it is used across pages of some features

File: lib/consumer/home/presentation/search_bar_filter.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import 'package:flutter/material.dart';
```
Comment (sinhaugoh) [inline lib/consumer/home/presentation/search_bar_filter.dart]: this should be a widget? can put inside a widget folder

File: lib/consumer/home/presentation/retailer_listview.dart [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        switch (snapshot.connectionState) {
+          case ConnectionState.waiting:
+            return const CircularProgressIndicator();
+          default:
+            if (snapshot.hasError) {
+              return Text('Error: ${snapshot.error}');
+            } else {
+              return ListView.builder(
+                itemCount: snapshot.data!.length,
+                itemBuilder: (context, index) => ProviderScope(
```
Comment (sinhaugoh) [inline lib/consumer/home/presentation/retailer_listview.dart]: i think the ProviderScope is redundant?

**Review by sinhaugoh — CHANGES_REQUESTED** [review]
You can consider these changes :)

**Comment by Kwang-Kian-Hui** [comment]
Changes from feedback 3: also added query of retrieving retailers only if visibility is on

File: analysis_options.yaml (hunk 1 above)
Comment (Kwang-Kian-Hui) [reply analysis_options.yaml]: Used to resolve an error previously but is now redundant, removed in later commits.

File: lib/consumer/home/presentation/drawer_widget.dart (hunk 2 above)
Comment (Kwang-Kian-Hui) [reply lib/consumer/home/presentation/drawer_widget.dart]: okay moved it over

File: lib/consumer/home/presentation/retailer_listview.dart (hunk 4 above)
Comment (Kwang-Kian-Hui) [reply lib/consumer/home/presentation/retailer_listview.dart]: yeap you are right, removed in later commits

File: lib/consumer/application/retailer_list_notifier.dart [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        cancelledOperation: (_) {
+          state = const RetailerListState.failure("Operation cancelled.");
+        },
+        unknown: (_) {
+          state = const RetailerListState.failure("Unknown error.");
+        },
+        orElse: () {
+          state = const RetailerListState.failure("Unknown error.");
+        },
+      );
```
Comment (sinhaugoh) [inline lib/consumer/application/retailer_list_notifier.dart:24]: seems like unknown and orelse return the same thing, maybe u can remove the unknown part?

**Review by sinhaugoh — CHANGES_REQUESTED** [review]

File: lib/consumer/application/retailer_list_notifier.dart (hunk 5 above)
Comment (Kwang-Kian-Hui) [reply lib/consumer/application/retailer_list_notifier.dart:24]: okay, updated it on the latest commit

**Review by sinhaugoh — APPROVED** [review]

**Pull request merged**